File: Florist.cpp
#include<iostream>
#include<algorithm>
using namespace std;
int main(){
    int n,k,sum=0;
    cin>>n>>k;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    sort(arr,arr+n);
    int c=0,s=1;
    for(int i=(n-1);i>=0;i--){
        if(c<k){
            sum= sum + arr[i]*s;

            c=c+1;

        }
        else{
            s=s+1;
            sum=sum+arr[i]*(s);

            c=0;
        }
    }
    cout<<sum<<endl;
}

File: Java/Florist.java
import java.io.*;
import java.util.Scanner;
import java.util.Arrays;
class florist{
    public static void main(String[]args){
        Scanner in= new Scanner(System.in);
        int n=in.nextInt();
        int arr[]=new int[n];
        int k=in.nextInt();
        for(int i=0;i<n;i++){
            arr[i]=in.nextInt();

        }
        Arrays.sort(arr);
        int c=1;
        int h= 0;
        int sum=0;
        for(int i=n-1;i>=0;i--){
          if(c<=k){
              sum=sum+(1+h)*arr[i];
              c++;
          }
          else{
              
              c=1;
              h++;
              sum=sum+(1+h)*arr[i];
              c=c+1;
          }
        }
       System.out.println(sum);
    }
}